services/wallet/collectibles/collection_data_db.go

```diff
@@ -246,8 +277,109 @@ func (o *CollectionDataDB) GetData(ids []thirdparty.ContractID) (map[string]thir
 				return nil, err
 			}
 
+			// Get socials from different table
+			c.Socials, err = getCollectionSocials(o.db, c.ID)
+			if err != nil {
+				return nil, err
+			}
+
 			ret[c.ID.HashKey()] = *c
 		}
 	}
 	return ret, nil
 }
+
+func (o *CollectionDataDB) HasValidSocialsForID(contractID thirdparty.ContractID) (bool, error) {
+	return hasSocialsForID(o.db, contractID)
+}
+
+func (o *CollectionDataDB) SetCollectionSocialsData(provider string, id thirdparty.ContractID, collectionSocials *thirdparty.CollectionSocials) (err error) {
+	tx, err := o.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		_ = tx.Rollback()
+	}()
+
+	// Insert new collections socials
+	hasSocials, err := hasSocialsForID(tx, id)
+	if err != nil {
+		return err
+	}
+
+	if !hasSocials && collectionSocials != nil {
```

Comment:
No need to check 'hasSocials' here IMO, just overwrite if there's new data. In any case the responsibility of this check should be on the caller's side